### ISSUE TITLE
Allow extension for the README file

### ISSUE THREAD

**Issue by mateuszpawlik** [bids-standard/bids-specification#972]
GitLab cannot render markdown if the file doesn't have an extension. Currently, BIDS Validator complains that there is no `README` file although `README.md` is there. I think this is due to the following rule:

https://github.com/bids-standard/bids-specification/blob/febae4fb9b2c9f055dc1392e0283c85f05618de0/src/schema/rules/top_level_files.yaml#L2:L5

Could we change that and support file extensions for the `README` file? I know it would require a few changes in the documentation and maybe the validator but if we could work out the list, I'm willing to work on it.

**Comment by VisLab** [comment]
This would be a welcome addition and might encourage annotators to write a better `README`.  Many people already use markdown for the `README` because OpenNeuro displays the file without the extension as markdown.

**Comment by sappelhoff** [comment]
I agree that these are good arguments to allow `.md` (and `.txt`) :+1:

**Comment by sappelhoff** [comment]
This also came up before https://github.com/bids-standard/bids-validator/pull/1281#discussion_r630118909

@effigies comment from back then shows that he'd also be in favor of supporting this (with a small caveat):

> I would be +1 for permitting markdown, at least. It opens up a smallish can of worms as we now get into the role of arbitrating the set of acceptable markup languages.

**Comment by mateuszpawlik** [comment]
> This also came up before [bids-standard/bids-validator#1281 (comment)](https://github.com/bids-standard/bids-validator/pull/1281#discussion_r630118909)

Thanks for finding it.

> > I would be +1 for permitting markdown, at least. It opens up a smallish can of worms as we now get into the role of arbitrating the set of acceptable markup languages.

I wouldn't go a far as limiting and verifying Markdown flavors. I think it is just about the file extension. As per specification, `README` file [is a free-form text](https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#readme). On the other hand, the [recommended README template](https://github.com/bids-standard/bids-starter-kit/blob/main/templates/README) uses Markdown already.

And if you would like to limit the Markdown flavor, I propose something standard, like [CommonMark](https://commonmark.org/), since both [GitHub](https://github.github.com/gfm/#what-is-github-flavored-markdown-) and [GitLab](https://docs.gitlab.com/ee/user/markdown.html#differences-between-gitlab-flavored-markdown-and-standard-markdown) flavors are its supersets.

**Comment by mateuszpawlik** [comment]
I'm happy to help and I have time to commit.

**Comment by sappelhoff** [comment]
Good to hear! I think you could start to open PRs then, and we can iterate there.

**Comment by sappelhoff** [comment]
@mateuszpawlik -- just a short reminder that I think your proposal is worthwhile implementing. Let me know if you ran out of time and want me /someone else to do it instead.

**Comment by mateuszpawlik** [comment]
I'm sorry but I didn't have a long enough stretch to figure out where to start and where exactly the changes should be implemented. I still would like to do it because it's a nice, simple enough exercise to understand the ecosystem. Maybe there are some contribution details that list the things to change? If not, I'll start making a list tomorrow. Would that work?

**Comment by sappelhoff** [comment]
These are two important documents to read:

- https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md
- https://github.com/bids-standard/bids-specification/blob/master/DECISION-MAKING.md

As for making the changes, that'd be in the specification text - probably somewhere here:

- https://github.com/bids-standard/bids-specification/blob/master/src/03-modality-agnostic-files.md

For the validator, you could open an issue on https://github.com/bids-standard/bids-validator/ and we can try to see where to put the check there. I wouldn't know that off the top of my head

**Comment by mateuszpawlik** [comment]
I've started working but failed already while building #1030 .
